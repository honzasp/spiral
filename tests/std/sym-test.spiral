(program
  (import std)
  (import std.test)

  (test "sym?" (lambda (t)
    (assert t (sym? (sym-new "pi")))
    (assert-not t (sym? (ref-new "pi")))
    (assert-not t (sym? "pi"))))

  (test "sym-get" (lambda (t)
    (let ((s0 (sym-new 1000))
          (s1 (sym-new "foo")))
      (assert-eqv t (sym-get s0) 1000)
      (assert-eqv t (sym-get s1) "foo")))))
