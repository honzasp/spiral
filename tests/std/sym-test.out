test sym? ok
test sym-get ok
